### DISCUSSION TITLE
How do I view or collect SqlClient Traces on Azure?

### DISCUSSION THREAD

**Discussion by Trivedi-Ashish** [dotnet/SqlClient#2169] [General]
How do I enable and view SqlClient traces if I have a web app hosted on Azure and it uses SqlClient. I know that we can use XPerf/PerfView to collect and view traces. But I am not sure how to use it when the web app is hosted on Azure app service.

**Comment by David-Engel** [comment]
Create your own listener in your app:
https://learn.microsoft.com/en-us/sql/connect/ado-net/enable-eventsource-tracing?view=sql-server-ver16#example

In OnEventWritten, you can log to your app service logs.

**Reply by ErikEJ** [reply]
I have a NuGet package that might be able to help you; https://www.nuget.org/packages/ErikEJ.SqlClient.Extensions

**Reply by pfdsilva** [reply]
@David-Engel Are SQL Server drivers integrated with **Microsoft.Extensions.Logging**?
(https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging?view=dotnet-plat-ext-7.0)

**Reply by ErikEJ** [reply]
@pfdsilva no, but my project shows a possible way to do just that
